### PR TITLE
RF+ENH: GitRepo.clone to wrap around clone operation and provide workarounds

### LINKED ISSUE #785: git clone http://  is unreliable
some times works some times doesn't :-/  git 2.9.3

```shell
(venv-tests) % rm -rf fbirn_phaseIII; git clone http://datasets.datalad.org/nidm/fbirn_phaseIII/.git
Cloning into 'fbirn_phaseIII'...
Checking connectivity... done.

(venv-tests) % rm -rf fbirn_phaseIII; git clone http://datasets.datalad.org/nidm/fbirn_phaseIII/.git
Cloning into 'fbirn_phaseIII'...
Checking connectivity... done.

(venv-tests) % rm -rf fbirn_phaseIII; git clone http://datasets.datalad.org/nidm/fbirn_phaseIII/.git
Cloning into 'fbirn_phaseIII'...
error: Request for d53302dfc7ad13b786923927021039d21a10d5bd aborted
error: Unable to find d53302dfc7ad13b786923927021039d21a10d5bd under http://datasets.datalad.org/nidm/fbirn_phaseIII/.git
Cannot obtain needed tree d53302dfc7ad13b786923927021039d21a10d5bd
while processing commit 22dd4c49417cad6f4082ac2aebef45da8b6e473d.
error: fetch failed.
```

so I wonder if we should provide a workaround for that :-/
I suspect that it tries to be efficient, and somewhere there is a race condition or may be server/firewall limit dropping... tested our firewall by stopping it -- still the same situation

### PR DESCRIPTION
**Pull request by yarikoptic**
Closes #785

### CODE REVIEW

**Comment by mih** [comment]
Holy cow.

File: datalad/support/gitrepo.py [hunk 1]
```diff
@@ -480,6 +463,51 @@ def __init__(self, path, url=None, runner=None, create=True,
...
+
+        Provides workarounds for known issues (e.g.
+        https://github.com/datalad/datalad/issues/785)
+
+        Parameters
+        ----------
+        url : str
+        path : str
+        """
+        ntries = 5  # r is not enough for robust workaround
```
Comment (yarikoptic) [inline datalad/support/gitrepo.py]: doh -- should have been '3' not 'r' in the comment ;)

**Comment by bpoldrack** [comment]
Nice.

**Comment by yarikoptic** [comment]
I would appreciate a quick review and merge @datalad/developers

**Pull request merged**